src/modules/style.ts

```diff
@@ -1,9 +1,13 @@
 import { VNode, VNodeData } from "../vnode";
 import { Module } from "./module";
 
-export type VNodeStyle = Record<string, string> & {
-  delayed?: Record<string, string>;
-  remove?: Record<string, string>;
+export type ElementStyle = {
+  [Property in keyof ElementCSSInlineStyle]?: ElementCSSInlineStyle[Property]
+}
```

Comment:
This doesn't seem right. The type `ElementCSSInlineStyle` only contains the keys `style` and `attributeStyleMap`. But what we want is the CSS style names themselves (`background`, `fontStyle`, etc.). It seems that `CSSStyleDeclaration` is the type we want instead. When I change the type to that I get auto-completions for the CSS style names.

Also, `Partial` can be used instead of the explicit map to create a partial version of an object.

Reply:
`ElementCSSInlineStyle` I thought I copied from working code, so I'll triple check it there too when I get a chance. I hope it is not something silly like different lib.dom versions, but it might be a silly mistake of a different sort.

Good catch on the `Partial<>`, that mapped type has an additional transformation in my original code, but in this form you are right is just Partial. I'll correct that.